src/core/raster/qgsrasterlayer.h

```diff
@@ -384,6 +384,23 @@ class CORE_EXPORT QgsRasterLayer : public QgsMapLayer
      */
     void refreshRendererIfNeeded( QgsRasterRenderer *rasterRenderer, const QgsRectangle &extent ) SIP_SKIP;
 
+    /**
+     * Returns the string (typically sql) used to define a subset of the layer.
+     * \returns The subset string or null QString if not implemented by the provider
+     * \since QGIS 3.12
+     */
+    virtual QString subsetString() const;
```

Comment:
Is it worth bouncing these up to QgsMapLayer? (I'm unsure, interested in your thoughts)

@PeterPetrik is it likely that mesh layers will someday support string filters?

Reply:
Yes, eventually.

Reply:
not sure if likely, but it could be implemented (not easy)..